firmware/src/main.cpp

```diff
@@ -20,11 +21,18 @@ static MotorTask motor_task(1, config);
 
 InterfaceTask interface_task(0, motor_task, display_task_p);
 
+#ifdef SECRETS_H
 WifiTask wifiTask(WifiSSID, WifiPassword);
+#else
+#error "secrets.h is missing. Please rename secrets.h.example to secrets.h and set your own secrets."
+#endif
 
 void setup()
 {
+
+#ifdef SECRETS_H
   wifiTask.initializeWifi(); 
+#endif
```

Comment:
could you please move the if else inside the wifi? think clean code, layer separation. Main doesn't need to know details about wifi implementation. It kicks off wifi and wifi task will decide the logic about what to do, what to log etc.